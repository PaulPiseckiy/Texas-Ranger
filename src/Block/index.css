.content:last-child {
  margin-bottom: 100px;
}

.content:hover {
  box-shadow: 0 0 10px 1px lightgreen;
  border-color: lightgreen;
}

.content > button {
  font-size: 15px; 
  background-color: white;
  border: 1px solid lightgrey;
  transition: all 0.15s ease-in-out;
}

.content > button:hover {
  background-color: lightgreen;
  border-color: lightgreen;
}

.content-close {
  height: 30px;
  width: 60px;
}

.content-tofav {
  height: 30px;
  width: 60px;
}
