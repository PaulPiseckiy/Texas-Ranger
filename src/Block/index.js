//@flow
import React from 'react';
import type {ReactChildren} from 'react-flow-types';
import './index.css'

const Block = ({ 
  children, 
  onClick, 
  onDismiss, 
  isFavouriteJoke
}: {
  children: ReactChildren,
  onClick: () => mixed,
  onDismiss: () => mixed,
  isFavouriteJoke: string
}) => 
  <div className="content">
      <button 
        className={`content-tofav ${isFavouriteJoke}`}
        onClick={onClick}
      >
        <i className="fa fa-star" />
      </button>
    <div>
      {children}
    </div>
    <button 
      className="content-close"
      onClick={onDismiss}
    >
      X
    </button>
  </div>

export default Block;
