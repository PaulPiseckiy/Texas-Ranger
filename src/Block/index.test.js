import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Block from '../Block';

describe('Block', () => {

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Block />, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <Block />
    );
    let tree = component.toJSON;
    expect(tree).toMatchSnapshot();
  });

});

