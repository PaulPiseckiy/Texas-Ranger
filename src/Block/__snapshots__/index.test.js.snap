// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block snapshots 1`] = `[Function]`;
